app/models/blood_pressure.rb

```diff
@@ -64,4 +64,8 @@ def anonymized_data
      bp_systolic: systolic,
      bp_diastolic: diastolic}
   end
+
+  def self.for_recent_bp_log
+    order(Arel.sql("DATE(recorded_at at time zone 'utc' at time zone '#{CountryConfig.current[:time_zone]}') DESC, recorded_at ASC"))
+  end
```

Comment:
Such a niche use case, I don't think the model's the right place for this. It's not a very widely used concept to order the BPs this way. Can we extract this to a query object?

```
RecentBPLogQuery.for(user: @user)

class RecentBPLogQuery
  def self.for(scope: true)
    BloodPressure.where(scope).order(the_fancy_stuff)
  end
end
```

We should also add some tight unit test coverage around the timezone boundaries and ordering for this, regardless of where it lives.

Reply:
That sounds fantastic👌🏼

Reply:
On second thought, a query object doesn't suit this use case since it just orders BPs more or less. Moved it to a view helper in 32e8df5